### ISSUE TITLE
Neural network construction

### ISSUE THREAD

**Issue by aikekeaishenghuo** [JRMeyer/multi-task-kaldi#3]
feat_dim=`feat-to-dim scp:${multi_data_dirs[0]}/feats.scp -`
    hidden_dim=$hidden_dim
    # The following definition is for the shared hidden layers of the nnet!
    cat <<EOF > $exp_dir/configs/network.xconfig
input dim=$feat_dim name=input
relu-renorm-layer name=tdnn1 input=Append(input@-2,input@-1,input,input@1,input@2) dim=$hidden_dim
relu-renorm-layer name=tdnn2 dim=$hidden_dim
---------------------------------------------------------------------------------
 I want to join in ivector here
For example, when creating a single task  -------------
  input dim=100 name=ivector
  input dim=43 name=input
  fixed-affine-layer name=lda input=Append(-2,-1,0,1,2,ReplaceIndex(ivector, t, 0)) affine-transform-file=$dir/configs/lda.mat
  # the first splicing is moved before the lda layer, so no splicing here
  relu-batchnorm-layer name=tdnn1 dim=850
----------------------------------------------------------------------------------
But I did not add successfully, I do not know how the network should be designed, can you give some advice

**Comment by QuadraV-Speech** [comment]
i have meet this issue also , the solution is that , modify the train-script  "train_raw_dnn.py" ,  add parameter "use_multitask_egs=True " to function "train_lib.common.compute_preconditioning_matrix"

**Comment by aikekeaishenghuo** [comment]
Thanks
I want to do both&nbsp;&nbsp;voiceprint recognize&nbsp;and speech recognition tasks together.
Can you give me some advice？&nbsp; &nbsp;
I don't know how to do it now.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Ephemeroptera"<notifications@github.com&gt;;
发送时间:&nbsp;2019年12月27日(星期五) 晚上7:56
收件人:&nbsp;"JRMeyer/multi-task-kaldi"<multi-task-kaldi@noreply.github.com&gt;;
抄送:&nbsp;"沐沐"<1534560830@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [JRMeyer/multi-task-kaldi] Neural network construction (#3)




i have meet this issue also , the solution is that , modify the train-script  "train_raw_dnn.py" ,  add parameter "use_multitask_egs=True " to function "train_lib.common.compute_preconditioning_matrix"

**Comment by JRMeyer** [comment]
this is a request for a specific use-case, rather than an issue with the code, so I'm closing the issue
